pkg/find/vulnerabilities/trivy/converter.go

```diff
@@ -2,56 +2,86 @@ package trivy
 
 import (
 	"encoding/json"
-	sec "github.com/aquasecurity/starboard/pkg/apis/aquasecurity/v1alpha1"
 	"io"
+	"io/ioutil"
+	"strings"
+
+	starboard "github.com/aquasecurity/starboard/pkg/apis/aquasecurity/v1alpha1"
 )
 
 // Converter is the interface that wraps the Convert method.
 //
 // Convert converts the vulnerabilities model used by Trivy
-// to a generic model defined by K8S-native security CRDs.
+// to a generic model defined by the Custom Security Resource Specification.
 type Converter interface {
-	Convert(reader io.Reader) (sec.VulnerabilityReport, error)
+	Convert(reader io.Reader) (starboard.VulnerabilityReport, error)
 }
 
 type converter struct {
 }
 
-var DefaultConverter Converter = &converter{}
+var DefaultConverter = NewConverter()
+
+func NewConverter() Converter {
+	return &converter{}
+}
 
-func (c *converter) Convert(reader io.Reader) (report sec.VulnerabilityReport, err error) {
+func (c *converter) Convert(reader io.Reader) (report starboard.VulnerabilityReport, err error) {
 	var scanReports []ScanReport
-	err = json.NewDecoder(reader).Decode(&scanReports)
+	skipReader, err := c.skippingNoisyOutputReader(reader)
+	if err != nil {
+		return
+	}
+	err = json.NewDecoder(skipReader).Decode(&scanReports)
 	if err != nil {
 		return
 	}
 	report = c.convert(scanReports)
 	return
 }
 
-func (c *converter) convert(reports []ScanReport) sec.VulnerabilityReport {
-	var vulnerabilities []sec.VulnerabilityItem
+// TODO Normally I'd use Trivy with the --quiet flag, but in case of errors it does suppress the error message.
+// TODO Therefore, as a workaround I do sanitize the input reader before we start parsing the JSON output.
+func (c *converter) skippingNoisyOutputReader(input io.Reader) (reader io.Reader, err error) {
```

Comment:
info: IMHO, we may not need to use named result parameters in this case.

>Don't name result parameters just to avoid declaring a var inside the function

>Corollary: it's not worth it to name result parameters just because it enables you to use naked returns.

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Reply:
Thank you for pointing this out. TBH I don't have strong opinions on that. I only saw lots of code in other projects using naked returns. But this code review wiki makes perfect sense to me.

Reply:
Removed names from the result params in skippingNoisyOutputReader method. I'll be fixing it in other places gradually in the subsequent PRs

Reply:
As the document says, it depends on the situation. In some cases, we should use named params/return, but it is not always the case.

Reply:
Yes. I agree.